### PR TITLE
Setup automation for kubernetes-csi

### PR DESCRIPTION
**Pull request by spiffxp**
ref: #8989

/cc @saad-ali
This will cooperate with the existing travis-ci stuff you have setup

/hold
For comment. Also not sure I want to deploy this on a Friday

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -155,6 +155,28 @@ branch-protection:
...
+            contexts:
+            - continuous-integration/travis-ci
+        external-attacher:
+          required_status_checks:
+            contexts:
+            - continuous-integration/travis-ci
+        external-provisioner:
+          required_status_checks:
+            contexts:
+            - continuous-integration/travis-ci
```
Comment (saad-ali) [inline prow/config.yaml]: Travis CI is also enabled in these repos:
* https://github.com/kubernetes-csi/driver-registrar
* https://github.com/kubernetes-csi/external-snapshotter
* https://github.com/kubernetes-csi/livenessprobe

Not sure if they should also be specified here.

Comment (spiffxp) [reply prow/config.yaml]: When I was looking at the repo settings, those contexts weren't yet listed as required for merge.  I'd rather keep things exactly as they are, and then step away and let you make any further changes you'd like via PR.

Comment (saad-ali) [reply prow/config.yaml]: Makes sense. SGTM!

**Comment by xing-yang** [comment]
I have made "passing travis-ci required before merging" for https://github.com/kubernetes-csi/external-snapshotter.  @saad-ali @spiffxp Can you please check and let me know if anything is missing?  Thanks!

**Pull request merged**